packages/strapi-plugin-upload/admin/src/components/FiltersPicker/FiltersCard/index.js

```diff
@@ -4,31 +4,33 @@ import { FormattedMessage } from 'react-intl';
 
 import { Select } from '@buffetjs/core';
 import { getFilterType } from 'strapi-helper-plugin';
-import getTrad from '../../utils/getTrad';
+import getTrad from '../../../utils/getTrad';
 
 import reducer, { initialState } from './reducer';
 
+import filters from './utils/filtersForm';
+
 import Wrapper from './Wrapper';
-import Button from './Button';
 import InputWrapper from './InputWrapper';
+import FilterButton from './FilterButton';
+import FilterInput from './FilterInput';
 
-import Input from '../FilterInput';
-
-const FiltersCard = ({ filters, onChange }) => {
+const FiltersCard = ({ onChange }) => {
   const [state, dispatch] = useReducer(reducer, initialState);
 
-  const name = state.get('name');
-  const type = filters[name].type;
-
-  const defaultValue = filters[name].defaultValue;
+  const { name, filter, value } = state.toJS();
 
+  const type = filters[name].type;
   const filtersOptions = getFilterType(type);
 
+  const getDefaultValue = name => filters[name].defaultValue;
+
   const handleChange = ({ target: { name, value } }) => {
     dispatch({
       type: 'ON_CHANGE',
       name,
       value,
+      defaultValue: filters[value] ? getDefaultValue(value) : null,
```

Comment:
This should be done in your reducer directly since you also have access to the name.